src/AccessibilityInsights.Extensions.GitHub/LinkValidator.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+using System;
+using System.Globalization;
+using System.Text.RegularExpressions;
+
+namespace AccessibilityInsights.Extensions.GitHub
+{
+    /// <summary>
+    /// GitHub Repo Link Format Validator
+    /// </summary>
+    public static class LinkValidator
+    {
+        private static readonly string GitHubLink = Properties.Resources.GitHubLink;
+        private static readonly string AlphaNumricPattern = Properties.Resources.AlphaNumricPattern;
+
+        public static bool IsValidGitHubRepoLink(string Link)
+        {
+            Link = Link.Replace(@"\", "/").Trim(' ').TrimEnd('/');
+            string UserNamePattern = string.Format(CultureInfo.InvariantCulture, Properties.Resources.UserNamePattern, AlphaNumricPattern);
+            string RepoNamePattern = string.Format(CultureInfo.InvariantCulture, Properties.Resources.RepoNamePattern, AlphaNumricPattern);
+            string LinkPatttern = string.Format(CultureInfo.InvariantCulture, Properties.Resources.LinkPatttern, GitHubLink, UserNamePattern, RepoNamePattern);
+            Regex gitHubRepoLinkRegex = new Regex(LinkPatttern, RegexOptions.IgnoreCase);
+            if (!gitHubRepoLinkRegex.IsMatch(Link))
+            {
+                return false;
+            }
+
+            Link = Regex.Replace(Link, GitHubLink, "");
+            string[] parts = Regex.Split(Link, @"/+");
+            if (parts.Length != 2)
+            {
+                return false;
+            }
+
+            string userName = parts[0];
+            string repoName = parts[1];
+            if (!CheckUserNameLength(userName) || !CheckUserNameAtLeastOneChar(userName) || !CheckRepoNameLength(repoName) || !CheckRepoNameSpecialCases(repoName))
+            {
+                return false;
+            }
+
+            return true;
```

Comment:
If this is supposed to be a fully formed Uri, then it might make sense to verify that it's not going to cause an Exception when we use it.
```suggestion
            return Uri.IsWellFormedString(link, UriKind.Absolute);
```

Reply:
I don't think it's necessary, we only check for regex